metro.config.js

```diff
@@ -0,0 +1,14 @@
+// Learn more https://docs.expo.io/guides/customizing-metro
+const { getDefaultConfig } = require('expo/metro-config');
+
+/** @type {import('expo/metro-config').MetroConfig} */
+const config = getDefaultConfig(__dirname);
+
+const defaultSourceExts = require('metro-config/src/defaults/defaults').sourceExts;
+
+if (process.env.EXPO_PUBLIC_DETOX_MOCKED === 'mocked'){
+  config.resolver.sourceExts = ['mock.tsx', ...defaultSourceExts];
+}
+
+
+module.exports = config;
```

Comment:
I think I prefer this syntax, maybe let's not depend on a fixed string "mocked" but rather `true`? But wouldn't we want detox to be always mocked? Maybe not let me know. 
https://github.com/graasp/graasp-mobile-builder/blob/48f5eaba1cbf9d62e25f9e28ebdbea7177da1b4e/metro.config.js

Reply:
I refactored it with your syntax, please take a look